system/lib/libc/musl/src/unistd/dup2.c

```diff
@@ -19,9 +16,11 @@ int dup2(int old, int new)
 		r = __syscall(SYS_fcntl, old, F_GETFD);
 #endif
 		if (r >= 0) return old;
-	} else {
-		while ((r=__syscall(SYS_dup3, old, new, 0))==-EBUSY);
 	}
+	// __EMSCRIPTEN__ removed else block here to avoid returning
+	// uninitilized `r` when old and new are equal but old is
+	// invalid.
+	while ((r=__syscall(SYS_dup3, old, new, 0))==-EBUSY);
```

Comment:
removed => moved perhaps? Not sure I follow the logic here.

Also I don't see how this avoids a use of uninitialized `r`. The old code assigns `r` in all code paths AFAICT: if `SYS_dup2`, if `old == new`, and if `old != new`.

Reply:
Oh I see, there was a bug above there r is assiged to ` __wasi_fd_is_valid(old) ? 0 : -1;` .. this is wrong.  I will need to fix.

Reply:
Found a better fix.